shapely/strtree.py

```diff
@@ -45,6 +45,11 @@ class STRtree:
 
     Any mixture of geometry types may be stored in the tree.
 
+    The tree automatically adds a write lock to the input geometry array to
+    prevent modification of geometries indexed in the tree.  You can avoid this
+    by passing a copy of the geometry array as input or by deleting the tree
```

Comment:
Should we mention that this is actually a relatively cheap operation? (passing `arr.copy()`) 

Copying a numpy object array only creates a new array of pointers to the Python objects, but the actual geometries are not duplicated.

Reply:
To illustrate:

```
In [8]: coords = np.random.randn(1_000_000, 2)

In [9]: arr = shapely.points(coords)

In [10]: %timeit shapely.points(coords)
176 ms ± 3.26 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [11]: %timeit arr.copy()
15.5 ms ± 248 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

So copying one million geometries takes 15ms (and here it are simple points, but it should take as long for complex geometries).

This is actually also faster than creating the tree:
```
In [16]: %timeit shapely.STRtree(arr)
253 ms ± 5.3 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [17]: %timeit shapely.STRtree(arr.copy())
275 ms ± 3.3 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

So should we maybe consider just copying ourselves under the hood? (instead of asking the user to do this)

Reply:
Thanks for the additional testing!  Given that it is fast enough and low-overhead memory wise, I agree that we can just do that automatically, and it sidesteps need to lock the array or ask the user to pass a copy / delete the tree.